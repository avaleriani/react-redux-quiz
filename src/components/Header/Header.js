import React from "react";
import styles from "./styles.module.css";

/**
 * @param {{ timer: string, currentScore: number, highestScore: number}} props
 * @return HTMLElement
 */
const Header = (props) => (
  <div className={styles['header']}>
    <div>{props.timeLeft}</div>
    <div>{props.currentScore}</div>
    <div>{props.highestScore}</div>
  </div>
);

export default Header;

