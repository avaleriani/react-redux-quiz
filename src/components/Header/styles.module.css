.header {
    display: flex;
    justify-content: space-between;
    height: 90px;
    background-color: rgba(37,77,50, 0.75);
    align-items: center;
    padding: 0 40px 0 40px;
}