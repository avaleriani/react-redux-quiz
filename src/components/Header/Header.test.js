import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header component', () => {

  test('renders correctly without props', () => {
    const HeaderComponent = renderer.create(<Header/>).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const props = {
      roundPoints: 2, timeLeft: 50, currentScore: 12,
      highestScore: 24, currentLevel: 2
    };
    const HeaderComponent = renderer.create(<Header {...props}/>).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });
});