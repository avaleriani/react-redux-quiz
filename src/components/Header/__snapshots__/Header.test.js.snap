// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly without props 1`] = `
<div
  className="header"
>
  <div>
    <span>
      <span
        aria-label="timer"
        role="img"
      >
        ⏲
      </span>
      ️ Time left: 
    </span>
    <span />
  </div>
  <div>
    <span>
      <span
        aria-label="joystick"
        role="img"
      >
        🕹️
      </span>
       Level: 
    </span>
    <span />
  </div>
  <div>
    <span>
      <span
        aria-label="joystick"
        role="img"
      >
        🎮
      </span>
       Current score: 
    </span>
    <span />
  </div>
  <div>
    <span>
      <span
        aria-label="a hundred"
        role="img"
      >
        💯 
      </span>
       Highest score: 
    </span>
    <span />
  </div>
  <div>
    <span>
      <span
        aria-label="dice"
        role="img"
      >
        🎲 
      </span>
       Round points: 
    </span>
    <span />
  </div>
</div>
`;

exports[`Header component renders with props 1`] = `
<div
  className="header"
>
  <div>
    <span>
      <span
        aria-label="timer"
        role="img"
      >
        ⏲
      </span>
      ️ Time left: 
    </span>
    <span>
      50
    </span>
  </div>
  <div>
    <span>
      <span
        aria-label="joystick"
        role="img"
      >
        🕹️
      </span>
       Level: 
    </span>
    <span>
      2
    </span>
  </div>
  <div>
    <span>
      <span
        aria-label="joystick"
        role="img"
      >
        🎮
      </span>
       Current score: 
    </span>
    <span>
      12
    </span>
  </div>
  <div>
    <span>
      <span
        aria-label="a hundred"
        role="img"
      >
        💯 
      </span>
       Highest score: 
    </span>
    <span>
      24
    </span>
  </div>
  <div>
    <span>
      <span
        aria-label="dice"
        role="img"
      >
        🎲 
      </span>
       Round points: 
    </span>
    <span>
      2
    </span>
  </div>
</div>
`;
